src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+class MachineServiceImpl implements MachineService<Machine> {
+    private BulldozerProducer bulldozerProducer = new BulldozerProducer();
+    private ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private TruckProducer truckProducer = new TruckProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machineList = new ArrayList<>();
+        if (type.equals(Bulldozer.class)) {
+            machineList.addAll(bulldozerProducer.get());
+        }
+        if (type.equals(Excavator.class)) {
+            machineList.addAll(excavatorProducer.get());
+        }
+        if (type.equals(Truck.class)) {
+            machineList.addAll(truckProducer.get());
+        }
+        return machineList;
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method. This will help to simplify the method and make it more readable.